lambdas/check-existing-identity/src/main/java/uk/gov/di/ipv/core/checkexistingidentity/CheckExistingIdentityHandler.java

```diff
@@ -355,63 +355,74 @@ private VerifiableCredentialBundle getVerifiableCredentials(
         var tacticalVcs = verifiableCredentialService.getVcs(userId);
 
         if (configService.enabled(EVCS_WRITE_ENABLED) || configService.enabled(EVCS_READ_ENABLED)) {
+            var bundle = getEvcsCredentialBundle(userId, evcsAccessToken, tacticalVcs);
+            if (bundle != null) {
+                return bundle;
+            }
+        }
+
+        return new VerifiableCredentialBundle(tacticalVcs, false, false);
+    }
+
+    private VerifiableCredentialBundle getEvcsCredentialBundle(
+            String userId, String evcsAccessToken, List<VerifiableCredential> tacticalVcs)
+            throws CredentialParseException, EvcsServiceException {
+        try {
             var evcsVcs =
                     evcsService.getVerifiableCredentialsByState(
                             userId, evcsAccessToken, CURRENT, PENDING_RETURN);
 
             // Use pending return vcs to determine identity if available
             var evcsIdentityVcs = evcsVcs.get(PENDING_RETURN);
             var isPendingEvcs = true;
-            var hasPartiallyMigratedVcs = false;
-            VerifiableCredentialBundle vcBundle = null;
             if (isNullOrEmpty(evcsIdentityVcs)) {
                 evcsIdentityVcs = evcsVcs.get(CURRENT);
                 isPendingEvcs = false;
             }
-            if (!isNullOrEmpty(evcsIdentityVcs)) {
-                vcBundle =
-                        new VerifiableCredentialBundle(
-                                configService.enabled(EVCS_READ_ENABLED)
-                                        ? evcsIdentityVcs
-                                        : tacticalVcs,
-                                true,
-                                isPendingEvcs);
-
-                hasPartiallyMigratedVcs =
-                        hasPartiallyMigratedVcs(
-                                tacticalVcs,
-                                evcsIdentityVcs,
-                                isPendingEvcs,
-                                vcBundle.isF2fIdentity());
 
+            // Check for partially migrated pending identity
+            var hasPartiallyMigratedVcs =
+                    hasPartiallyMigratedVcs(tacticalVcs, evcsIdentityVcs, isPendingEvcs);
+            logIdentityMismatches(tacticalVcs, evcsVcs, hasPartiallyMigratedVcs);
+
+            if (!isNullOrEmpty(evcsIdentityVcs)) {
                 if (hasPartiallyMigratedVcs) {
                     // use tactical vcs but with evcs flags so that the store-identity lambda is
                     // called next and updates the evcs pending one
-                    vcBundle = new VerifiableCredentialBundle(tacticalVcs, true, true);
+                    return new VerifiableCredentialBundle(tacticalVcs, true, true);
+                } else {
+                    return new VerifiableCredentialBundle(
+                            configService.enabled(EVCS_READ_ENABLED)
```

Comment:
if EVCS_WRITE_ENABLED is removed then we don't need this check.

Reply:
Not quite - this check would still need to be `EVCS_READ_ENABLED` while the one above would be simplified to `EVCS_WRITE_ENABLED`.

I'm not sure it's that valuable to simplify the double condition though - if we can easily support both combinations in a rational way, why wouldn't we?

Reply:
We will only reach here for evcs VCs when EVCS_READ_ENABLED

Reply:
At the moment that's not the case, we're still reading from the tactical store for a couple of reasons:
- checking that we're correctly keeping them in sync (i.e. `logIdentityMismatches`)
- knowing whether to migrate the tactical store identity

We aren't using at as the source of truth though, which is what this check here is for (so that if EVCS reads are not enabled, we use tactical VCs regardless)

We could skip some of these checks, but we'd need to read from EVCS anyway to safely perform the migration, so it doesn't save us loads.